autoload/ledger.vim

```diff
@@ -400,3 +401,231 @@ func! ledger#entry()
+"
+" Returns 0 if the quickfix window is empty, 1 otherwise.
+function! s:quickfix_toggle(...)
+  if g:ledger_use_location_list
+    let l:list = 'l'
+    let l:open = (len(getloclist(winnr())) > 0)
+  else
+    let l:list = 'c'
+    let l:open = (len(getqflist()) > 0)
+  endif
+
+  if l:open
+    execute (g:ledger_qf_vertical ? 'vert' : 'botright') l:list.'open' g:ledger_qf_size
+    " Note that the following settings do not persist (e.g., when you close and re-open the quickfix window).
+    " See: http://superuser.com/questions/356912/how-do-i-change-the-quickix-title-status-bar-in-vim
+    if g:ledger_qf_hide_file
+      set conceallevel=2
+      set concealcursor=nc
+      syntax match qfFile /^[^|]*/ transparent conceal
+    endif
+    if a:0 > 0
+      let w:quickfix_title = a:1
+    endif
+    return 1
+  endif
+
+  execute l:list.'close'
+  call s:warning_message((a:0 > 1) ? a:2 : 'No results')
+  return 0
+endf
+
+" Populate a quickfix/location window with data. The argument must be a String
+" or a List.
+function! s:quickfix_populate(data)
+  " Note that cexpr/lexpr always uses the global value of errorformat
+  let l:efm = &errorformat  " Save global errorformat
+  set errorformat=%EWhile\ parsing\ file\ \"%f\"\\,\ line\ %l:,%ZError:\ %m,%-C%.%#
+  set errorformat+=%tarning:\ \"%f\"\\,\ line\ %l:\ %m
+  " Format to parse command-line errors:
+  set errorformat+=Error:\ %m
+  set errorformat+=%f:%l\ %m
+  set errorformat+=%-G%.%#
+  execute (g:ledger_use_location_list ? 'l' : 'c').'getexpr' 'a:data'
+  let &errorformat = l:efm  " Restore global errorformat
+  return
+endf
+
+" Parse a list of ledger arguments and build a ledger command ready to be
+" executed.
+"
+" Note that %, # and < *at the start* of an item are expanded by Vim. If you
+" want to pass such characters to Ledger, escape them with a backslash.
+"
+" See also http://vim.wikia.com/wiki/Display_output_of_shell_commands_in_new_window
+" See also https://groups.google.com/forum/#!topic/vim_use/4ZejMpt7TeU
+function! s:ledger_cmd(arglist)
+  let l:cmd = g:ledger_bin
+  for l:part in a:arglist
+    if l:part =~ '\v^[%#<]'
+      let l:expanded_part = expand(l:part)
+      let l:cmd .= ' ' . (l:expanded_part == "" ? l:part : shellescape(l:expanded_part))
+    else
+      let l:cmd .= ' ' . l:part
+    endif
+  endfor
+  return l:cmd
+endf
+
+function! s:is_ledger_buffer()
+  if getbufvar(winbufnr(winnr()), "&ft") !=# "ledger"
+    call s:error_message("Please switch to a Ledger buffer first.")
+    return 0
+  endif
+  return 1
+endf
+" }}}
+
+" Run an arbitrary ledger command to process the current buffer, and show the
+" output in a new buffer. If there are errors, no new buffer is opened: the
+" errors are displayed in a quickfix window instead.
+"
+" Parameters:
+" args  A string of Ledger arguments.
+function! ledger#report(args)
+  if !s:is_ledger_buffer() | return | endif
+  let l:cmd = s:ledger_cmd(['-f', '%'] + split(a:args, ' '))
```

Comment:
I think it would be useful if the `-f` argument is not hardcoded here. For example I edit `2015.ldg` but run my reports on `main.ldg`, which includes `2015.ldg` and may contain statements necessary for `2015.ldg` to balance. (I have `--file …/main.ldg` in my `~/.ledgerrc`.)